en/tutorials-and-examples/cms/authentication-plugin.rst

```diff
@@ -0,0 +1,288 @@
+CMS Tutorial - Authentication using cakephp/authentication
+##########################################################
+
+CakePHP now supports Authentication and Authorization features using the
+core plugins cakephp/authentication and cakephp/authorization.
+
+In this tutorial we are going to ensure passwords are stored securely in
+our database, we are going to provide a working login page, and apply some
+basic access control to the article creation & editing experiences.
+
+We will assume you have created the ``Controller``, ``Table``, ``Entity`` and
+templates for the ``users`` table in your database, you can do this manually
+like you did before for the ArticlesController, or you can use the bake shell
+to generate the classes for you using ``bin/cake bake all users``.
+
+Adding Password Hashing
+-----------------------
+
+If you were to create/update a user at this point in time, you might notice that
+the passwords are stored in plain text. This is really bad from a security point
+of view, so lets fix that.
+
+This is also a good time to talk about the model layer in CakePHP. In CakePHP,
+we separate the methods that operate on a collection of objects, and a single
+object into different classes. Methods that operate on the collection of
+entities are put in the ``Table`` class, while features belonging to a single
+record are put on the ``Entity`` class.
+
+For example, password hashing is done on the individual record, so we'll
+implement this behavior on the entity object. Because we want to hash the
+password each time it is set, we'll use a mutator/setter method. CakePHP will
+call convention based setter methods any time a property is set in one of your
+entities. Let's add a setter for the password. In **src/Model/Entity/User.php**
+add the following::
+
+    <?php
+    namespace App\Model\Entity;
+
+    use Cake\Auth\DefaultPasswordHasher; // Add this line
```

Comment:
Should they use the password hasher from the plugin?

Reply:
Absolutely